src/com/amazon/ion/benchmark/WriteRandomIonValues.java

```diff
@@ -341,13 +354,14 @@ private static void writeDataToFile(IonWriter writer, IonStruct constraintStruct
      */
     public static String constructString(IonStruct constraintStruct) throws Exception {
         String constructedString;
+        Integer codePointsLengthBound;
         String regexPattern = IonSchemaUtilities.parseTextConstraints(constraintStruct, IonSchemaUtilities.KEYWORD_REGEX);
-        Integer codePointsLengthBound = IonSchemaUtilities.parseConstraints(constraintStruct, IonSchemaUtilities.KEYWORD_CODE_POINT_LENGTH);
         if (regexPattern != null) {
             RgxGen rgxGen = new RgxGen(regexPattern);
             constructedString = rgxGen.generate();
         } else {
             Random random = new Random();
+            codePointsLengthBound = IonSchemaUtilities.parseConstraints(constraintStruct, IonSchemaUtilities.KEYWORD_CODE_POINT_LENGTH);
```

Comment:
What if there's both a regex and a code point length constraint?

Reply:
> What if there's both a regex and a code point length constraint?

In the following commit, the logic of constructing string will be categorized into three conditions:
1. ISL contains both `regex` and `codepoint_length`:
   First, generating string based on the provided `regex` then check whether the generated string aligned with the constraint `codepoint_length`. If not, the method will keep generating until the generated string matches both constraints. It might take a while to generate the string which matches two constraints at the same time or the provided constraints are conflict with each other (e.g. regex: "^B[0-9]{9}$", codepoint_length: range:: [1, 3]). Under this consideration, preset the times of implementing generating string step at most 10. When the iterations hit 10 times but not find any matches string, it will write null value.
2. Only contain one of the constraints:
   Construct the string conformed with the provided constraint.
3. No constraint:
   Under this condition, the method will construct string randomly with random Unicode codepoint and `codepoint_length`.

Note:
1. The upper bound of iteration time requires further discussion.
2. The unit test cannot be implemented until this issue be closed